### PR TITLE
feat: add enkrypt wallet

### PR DESCRIPTION
**Pull request by jmjles**

### CODE REVIEW

**Comment by DanielSinclair** [comment]
Enkrypt is not available as a WalletConnect product, correct? That logic should be stripped. Am also seeing issues with injection detection.
Screenshot 2023-01-27 at 9 00 59 AM

File: packages/rainbowkit/src/wallets/walletConnectors/enkryptWallet/enkryptWallet.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  const isEnkryptInjected =
+    typeof window !== 'undefined' &&
+    Boolean(
+      (
+        window.ethereum as typeof window.ethereum &
+          (undefined | { isEnkrypt?: boolean })
+      )?.isEnkrypt
+    );
+
+  const shouldUseWalletConnect = isMobile() && !isEnkryptInjected;
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/enkryptWallet/enkryptWallet.ts:26]: Unnecessary unless a mobile WalletConnect version is available

File: packages/rainbowkit/src/wallets/walletConnectors/enkryptWallet/enkryptWallet.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      const getUri = async () => {
+        const uri = (await connector.getProvider()).connector.uri;
+        return isMobile()
+          ? uri
+          : `https://enkrypt.com/wc?uri=${encodeURIComponent(uri)}`;
+      };
+
+      return {
+        connector,
+        mobile: {
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/enkryptWallet/enkryptWallet.ts:55]: Should be removed

File: packages/rainbowkit/src/wallets/walletConnectors/enkryptWallet/enkryptWallet.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    name: 'Enkrypt',
+    iconUrl: async () => (await import('./enkryptWallet.svg')).default,
+    iconBackground: '#fff',
+    installed: !shouldUseWalletConnect ? isEnkryptInjected : undefined,
+    downloadUrls: {
+      browserExtension:
+        'https://chrome.google.com/webstore/detail/enkrypt-ethereum-polkadot/kkpllkodjeloidieedojogacfhpaihoh',
+    },
+    createConnector: () => {
+      const connector = shouldUseWalletConnect
```
Comment (DanielSinclair) [inline packages/rainbowkit/src/wallets/walletConnectors/enkryptWallet/enkryptWallet.ts:39]: Should just return the injected connector here

**Review by DanielSinclair — CHANGES_REQUESTED** [review]

**Comment by gamalielhere** [comment]
@DanielSinclair fixed the corrections. Also couldn't replicate the injection issue on my end.

**Comment by DanielSinclair** [comment]
Still seeing issues on our example dApp deployment: https://rainbowkit-example-73h725g9i-rainbowdotme.vercel.app/
1. Injection not detected
Screenshot 2023-02-01 at 5 52 43 PM
2. Multiple connect attempt pop-ups appear on page load and ignore rejects
Screenshot 2023-02-01 at 5 52 08 PM

**Comment by DanielSinclair** [comment]
Closing for now, but can reopen and revisit if the provider and UX becomes more reliable. Working with teams like Phantom, Zerion, and Trust on this and happy to communicate further

**Comment by jayohTM** [comment]
> Closing for now, but can reopen and revisit if the provider and UX becomes more reliable. Working with teams like Phantom, Zerion, and Trust on this and happy to communicate further

Hi Daniel, I've been tracking this issue and would like to know more about the other wallet integrations.

**Pull request closed without merging**